modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -93,76 +121,46 @@ def apply(self, func, *args, **kwargs):
         -----
         It does not matter if `func` is callable or an ``ray.ObjectRef``. Ray will
         handle it correctly either way. The keyword arguments are sent as a dictionary.
+
+        If ``LazyExecution`` is enabled, the function is not applied immediately,
+        but is added to the execution tree.
         """
+        de = DeferredExecution(self._data_ref, func, args, kwargs)
+        if LazyExecution.get():
+            return self.__constructor__(de)
```

Comment:
@AndreyPavlenko when and in what form do you plan to add tests when `LazyExecution` is enabled?

Reply:
I think, we shouldn't overload the unit testing, but instead schedule weekly runs with this option enabled.